### ISSUE TITLE
Function name not displayed in google search results

### ISSUE THREAD

**Issue by benniledl** [WordPress/wporg-developer#479]
Hey, when I google: _wp_filter_build_unique_id
for some reason the function name is not in the title, its just this:

![Screenshot 2024-01-18 at 13-54-35 _wp_filter_build_unique_id - Google Suche](https://github.com/WordPress/wporg-developer/assets/109149472/6908ce90-ef3d-4990-8224-9d40dcb87857)

This shouldn't happen, other functions work fine:

![Screenshot 2024-01-18 at 13-55-05 get_post - Google Suche](https://github.com/WordPress/wporg-developer/assets/109149472/30ee28da-f3b8-4dd8-a23a-481b6ba864d6)

I am not sure on which pages this applies but maybe someone has a thought on why this happens?

**Comment by benniledl** [comment]
I think maybe this generally affects functions with a _ prefix
this is _custom_background_cb for example
![Screenshot 2024-01-18 at 14-17-38 _custom_background_cb - Google Suche](https://github.com/WordPress/wporg-developer/assets/109149472/2d95f2bb-3854-4686-9b53-6a3274978583)

**Comment by ryelle** [comment]
The `title` tag should include the function in both cases, so it's not a problem with the site. I wonder if Google thinks it's not a real part of the title.

We could switch up the code reference entry titles to prefix with the reference type, like "Function: _the_function_name() | Developer.WordPress.org" ? That way it wouldn't start with the underscore. 

@eidolonnight @thetinyl What do you think?

**Comment by eidolonnight** [comment]
I don't think it's the underscore that's a problem. Here is an example from the Drupal project docs:  [_forum_icon | forum.module | Drupal 4.6 | Drupal API](https://api.drupal.org/api/drupal/modules%21forum.module/function/_forum_icon/4.6.x)

![CleanShot 2024-01-18 at 15 30 59@2x](https://github.com/WordPress/wporg-developer/assets/115154/887dc442-dafa-4966-85a1-ee7677d36f68)

This does appear to be happening with other pages though, like [_wp_cron](https://developer.wordpress.org/reference/functions/_wp_cron/).

**Comment by ryelle** [comment]
True, when I search for `_wp_cron`, it appears in the second result's title.

<img width="776" alt="Screenshot 2024-01-18 at 3 39 01 PM" src="https://github.com/WordPress/wporg-developer/assets/541093/3c6ce76f-49d5-4302-ad32-e5d70b4f88b1">

But I'm not sure what we're doing differently, because we have the full title in `<title>` (and `og:title`, if that's worth anything here).

<img width="807" alt="Screenshot 2024-01-18 at 3 39 35 PM" src="https://github.com/WordPress/wporg-developer/assets/541093/c8230ce5-71f3-46d1-8653-0169d9abe24a">

**Comment by eidolonnight** [comment]
Were changes recently made to these pages? Perhaps it just needs to be re-indexed. Everything is looking fine currently.

**Comment by benniledl** [comment]
Maybe Google becomes confused due to the presence of multiple different titles, thinking it's inconsistent and it needs to figure out the title itself. It seems to extract the page name "Function" (stripping the seemingly wrong string starting with _) from the `<title>` tag and then derives the site name from other elements such as `<link rel="alternate">`, combining them, trying to take it's best guess. Maybe if we take the exact same title in every element, google will "trust" the title and not try to make its own guess?? 


![Screenshot 2024-01-18 222939](https://github.com/WordPress/wporg-developer/assets/109149472/3468eb0d-33af-45a5-be09-eadb0203406e)

For example this site: https://wp-kama.com/function/_wp_cron
Has the exact same string in every title element.

**Comment by ryelle** [comment]
> Were changes recently made to these pages?

There was https://github.com/WordPress/wporg-developer/pull/399, but it didn't change the title on these pages.

One thing I noticed while looking is that Google is changing the title slightly. It has the sections separated by `-`, rather than the `|`. For example, the page has `<title>get_theme() – Function | Developer.WordPress.org</title>`, but google shows:

<img width="652" alt="Screenshot 2024-01-18 at 4 47 27 PM" src="https://github.com/WordPress/wporg-developer/assets/541093/385deb18-3a75-4631-a1c8-788d15d5bbfb">

Oh, the suffix is different too- I just caught it because of @benni1516's screenshot (thanks!). It uses that `site_name` value for the last segment, "WordPress Developer Resources" rather than "Developer.WordPress.org".

So google is definitely adjusting the title. Maybe it sees the `h1` on the page is "_wp_cron(): int|false" and decided that isn't a human-friendly string? I'm not sure why the Drupal & wp-kama sites work, though, if that's the case.

**Comment by benniledl** [comment]
I think it strips _wp_cron() - because it is not a human-friendly string (as you said) and then it rather takes the sitename WordPress Developer Resources rather than the domain name because that is also a more human-friendly string.
